### ISSUE TITLE
Problem with "react-bootstrap/Col" in react-bootstrap 2.0.0-rc.0

### ISSUE THREAD

**Issue by simon-katz** [thheller/shadow-cljs#948]
I'm trying to upgrade from react-bootstrap 1.6.3 to 2.0.0-rc.0, and I've run into a problem or two.

This is a distillation of a previous discussion on Slack at https://clojurians.slack.com/archives/C0620C0C8/p1634210926174700 and https://clojurians.slack.com/archives/C0620C0C8/p1634212661178000

I’ve created a minimal reproduction at https://github.com/simon-katz/react-bootstrap-col-problem

In the `ui` namespace, there’s a var called demo-col-error? If false, the UI displays with no error. If true, I get the following error:

```
Uncaught TypeError: $jscomp.makeIterator is not a function
    at eval (Col.js:66)
    at renderWithHooks (react-dom.development.js:14986)
    at updateForwardRef (react-dom.development.js:17045)
    at beginWork (react-dom.development.js:19099)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3946)
    at Object.invokeGuardedCallbackImpl (react-dom.development.js:3995)
    at invokeGuardedCallback (react-dom.development.js:4057)
    at beginWork$1 (react-dom.development.js:23965)
    at performUnitOfWork (react-dom.development.js:22780)
    at workLoopSync (react-dom.development.js:22708)
```

A second problem (not sure if this is related): If I remove `["react-bootstrap/Col" :default rb-Col]` from the `(:require ...)` form, and also remove the things that use `rb-Col`, then initial from-scratch compilation is fine, but if I then re-introduce the require of "react-bootstrap/Col" incremental recompilation gives the following error:

```
NullPointerException: NAME $jscomp 66:8  [length: 96] [source_file: node_modules/react-bootstrap/cjs/Col.js]
```

For reference, here's the code I'm talking about:

```
(ns ui
  (:require
   ["react-bootstrap/Col"       :default rb-Col]
   ["react-bootstrap/Container" :as rb-Container]
   ["react-bootstrap/Row"       :as rb-Row]
   [reagent.core :as r]
   [reagent.dom :as dom]))

(def Col       (r/adapt-react-class rb-Col))
(def Container (r/adapt-react-class rb-Container))
(def Row       (r/adapt-react-class rb-Row))

(def demo-col-error? false)

(defn ^:dev/after-load render []
  (dom/render (if demo-col-error?
                ;; This gives: Uncaught TypeError: $jscomp.makeIterator is not a function
                [Container
                 [Row
                  [Col "1 of 1"]]]
                ;; No error with this:
                [Container
                 [Row "1"]])
              (js/document.getElementById "app")))
```

**Comment by svdo** [comment]
I don't think this was solved completely by the above commit. I created a repo that reproduces the issue for me: https://github.com/svdo/shadow-cljs-948. The README provides instructions on how I'm running things.

**Comment by simon-katz** [comment]
I also think the problems are only partially fixed.

I still have the first problem ("Uncaught TypeError: $jscomp.makeIterator is not a function").
Note: In my reproduction, you have to change the value of `demo-col-error?` to `true` to get this error.

The second problem ("NullPointerException") is fixed. (Thanks!)

**Comment by thheller** [comment]
I tracked it down to this issue https://github.com/google/closure-compiler/issues/3890.

In shadow-cljs `2.16.4` I added a (hopefully temporary) workaround that lets you specify which polyfills to inject manually. Adding ` :js-options {:force-library-injection #{"es6/util/makeiterator"}}` to your build config should fix this particular issue until the automatic detection works. That will ensure the polyfill is available and the code appears to work fine.
